src/python/pants/backend/nfpm/target_types_rules.py

```diff
@@ -0,0 +1,185 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from pants.backend.nfpm.fields.contents import (
+    NfpmContentDirDstField,
+    NfpmContentDirsField,
+    NfpmContentDstField,
+    NfpmContentFilesField,
+    NfpmContentFileSourceField,
+    NfpmContentSrcField,
+    NfpmContentSymlinkDstField,
+    NfpmContentSymlinksField,
+    NfpmContentSymlinkSrcField,
+)
+from pants.backend.nfpm.target_types import (
+    NfpmContentDir,
+    NfpmContentDirs,
+    NfpmContentFile,
+    NfpmContentFiles,
+    NfpmContentSymlink,
+    NfpmContentSymlinks,
+)
+from pants.engine.rules import collect_rules, rule
+from pants.engine.target import GeneratedTargets, GenerateTargetsRequest, InvalidFieldException
+from pants.engine.unions import UnionMembership, UnionRule
+from pants.util.logging import LogLevel
+from pants.util.strutil import softwrap
+
+
+class GenerateTargetsFromNfpmContentFilesRequest(GenerateTargetsRequest):
+    generate_from = NfpmContentFiles
+
+
+@rule(
+    desc="Generate `nfpm_content_file` targets from `nfpm_content_files` target",
+    level=LogLevel.DEBUG,
+)
+async def generate_targets_from_nfpm_content_files(
+    request: GenerateTargetsFromNfpmContentFilesRequest,
+    union_membership: UnionMembership,
+) -> GeneratedTargets:
```

Comment:
Possibly a RuleRunner based integration test... But, I don't see tests specifically for GeneratedTargets rules in other backends. I probably just need to include `nfpm_content_files(...)` in at least one of the tests for the `package_nfpm_package` rule.